### PR TITLE
Documentation -- Function Headers -- sys_prc343

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Handle documentation issue #107: sys_prc343 component. WIP.

### CODE REVIEW

File: addons/sys_prc343/radio/fnc_closeGui.sqf [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID (Unused) <STRING>
+ * 1: Event with the value "getChannelData" (Unused) <STRING>
+ * 2: Event data with the channel number (Unused) <NUMBER>
+ * 3: Radio data (Unused) <HASH>
  *
  * Return Value:
- * RETURN VALUE <TYPE>
+ * true <BOOL>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_closeGui.sqf]: `True <BOOL>`

File: addons/sys_prc343/radio/fnc_getCurrentChannel.sqf [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID <STRING>
+ * 1: Event (Unused) <STRING>
+ * 2: Event data (Unused) <NUMBER>
+ * 3: Radio data <HASH>
  *
  * Return Value:
- * RETURN VALUE <TYPE>
+ * Current channel id <NUMBER>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getCurrentChannel.sqf]: `ID`

File: addons/sys_prc343/radio/fnc_getCurrentChannel.sqf [hunk 3]
```diff
@@ -1,22 +1,24 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Returns the current channel of the active radio.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID <STRING>
+ * 1: Event (Unused) <STRING>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getCurrentChannel.sqf]: `<STRING> (Unused)` move unused to end.

File: addons/sys_prc343/radio/fnc_getCurrentChannel.sqf [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
+ * 2: Event data (Unused) <NUMBER>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getCurrentChannel.sqf]: Move unused to end.

File: addons/sys_prc343/radio/fnc_getChannelData.sqf [hunk 5]
```diff
@@ -1,24 +1,26 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Returns all information regarding the parsed radio channel.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID (Unused) <STRING>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getChannelData.sqf]: Move unused to end.

File: addons/sys_prc343/radio/fnc_getChannelData.sqf [hunk 6]
```diff
@@ -1,24 +1,26 @@
...
+ * 1: Event with the value "getChannelData" (Unused) <STRING>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getChannelData.sqf]: Move unused to end.

File: addons/sys_prc343/radio/fnc_getCurrentChannel.sqf [hunk 7]
```diff
@@ -1,22 +1,24 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Returns the current channel of the active radio.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID <STRING>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getCurrentChannel.sqf]: Also unused.

File: addons/sys_prc343/radio/fnc_getChannelDescription.sqf [hunk 8]
```diff
@@ -1,27 +1,26 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Returns the description of the currently selected channel. Used in the transmission hint.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID <STRING>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getChannelDescription.sqf:6]: Add the unused ones.

File: addons/sys_prc343/radio/fnc_closeGui.sqf [hunk 9]
```diff
@@ -1,16 +1,18 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Close radio GUI when calling this function. Event rised by onUnload (PRC343_RadioDialog).
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Radio ID (Unused) <STRING>
+ * 1: Event with the value "getChannelData" (Unused) <STRING>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_closeGui.sqf]: Move unused to end.

File: addons/sys_prc343/radio/fnc_closeGui.sqf [hunk 10]
```diff
@@ -1,16 +1,18 @@
...
+ * 2: Event data with the channel number (Unused) <NUMBER>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_closeGui.sqf]: Move unused to end.

File: addons/sys_prc343/radio/fnc_closeGui.sqf [hunk 11]
```diff
@@ -1,16 +1,18 @@
...
+ * 3: Radio data (Unused) <HASH>
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_closeGui.sqf]: Move unused to end.

File: addons/sys_prc343/radio/fnc_closeGui.sqf [hunk 12]
```diff
@@ -1,16 +1,18 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Close radio GUI when calling this function. Event rised by onUnload (PRC343_RadioDialog).
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_closeGui.sqf]: `raised`, no `when calling this function`, that's rather obvious.

**Review by jonpas — CHANGES_REQUESTED** [review]
No semi-colons in header examples.

**Comment by TheMagnetar** [comment]
Sorry I should have uploaded the changes.

File: addons/sys_prc343/functions/fnc_onChannelKnobPress.sqf [hunk 13]
```diff
@@ -10,7 +10,7 @@
  * RETURN VALUE <TYPE>
  *
  * Example:
- * [ARGUMENTS] call acre_COMPONENT_fnc_FUNCTIONNAME
+ * [ARGUMENTS] call acre_sys_prc343_fnc_onChannelKnobPress;
```
Comment (jonpas) [inline addons/sys_prc343/functions/fnc_onChannelKnobPress.sqf]: Still no semi-colons. :)

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/sys_prc343/functions/fnc_render.sqf [hunk 14]
```diff
@@ -1,16 +1,15 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Renders the radio when opened
```
Comment (jonpas) [inline addons/sys_prc343/functions/fnc_render.sqf]: Missing final dot.

File: addons/sys_prc343/functions/fnc_render.sqf [hunk 15]
```diff
@@ -1,16 +1,15 @@
...
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Display identifier <NUMBER>
  *
  * Return Value:
- * RETURN VALUE <TYPE>
+ * True <BOOL>
  *
  * Example:
- * [ARGUMENTS] call acre_sys_prc343_fnc_render;
+ * [DisplazID] call acre_sys_prc343_fnc_render
```
Comment (jonpas) [inline addons/sys_prc343/functions/fnc_render.sqf]: Y instead of Z.

File: addons/sys_prc343/functions/fnc_zoomChannelBlockSelector.sqf [hunk 16]
```diff
@@ -1,16 +1,16 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Triggers an animation when zooming in or out of the channel block selector
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
+ * 0: Zoom in(0) or out (1) <BOOL>
```
Comment (jonpas) [inline addons/sys_prc343/functions/fnc_zoomChannelBlockSelector.sqf]: Space after `in`.

File: addons/sys_prc343/radio/fnc_getCurrentChannelInternal.sqf [hunk 17]
```diff
@@ -1,19 +1,15 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Returns the unique identity of the current channel since the PRC343 has multiple blocks
```
Comment (jonpas) [inline addons/sys_prc343/radio/fnc_getCurrentChannelInternal.sqf]: Missing final dot.

**Review by jonpas — CHANGES_REQUESTED** [review]

**Comment by TheMagnetar** [comment]
I will test it later in the day since this PR also introduces params of the form: `params ["_radioId", "",  "", "", ""];`

**Comment by TheMagnetar** [comment]
I tested this and it should be ready to be merged.

**Review by jonpas — APPROVED** [review]

**Pull request merged**